src/IceRpc.Protobuf/MessageExtensions.cs

```diff
@@ -2,28 +2,30 @@
 
 using Google.Protobuf;
 using System.Buffers;
+using System.Buffers.Binary;
 using System.IO.Pipelines;
 
 namespace IceRpc.Protobuf;
 
-/// <summary>Provides extension methods for <see cref="IMessage" />.</summary>
+/// <summary>Provides an extension method for <see cref="IMessage" />.</summary>
 public static class MessageExtensions
 {
-    /// <summary>Merges data from a <see cref="PipeReader" /> into an existing <see cref="IMessage" />.</summary>
-    /// <param name="message">The <see cref="IMessage" /> to merge the data into.</param>
-    /// <param name="reader">The <see cref="PipeReader" /> containing the protobuf-encoded binary data to merge.</param>
-    public static async Task MergeFromAsync(this IMessage message, PipeReader reader)
-    {
-        using var stream = new MemoryStream();
-        await reader.CopyToAsync(stream).ConfigureAwait(false);
-        stream.Seek(0, SeekOrigin.Begin);
-        message.MergeFrom(stream);
-    }
-
-    /// <summary>Converts an <see cref="IMessage" /> into a <see cref="PipeReader" /> containing the protobuf-encoded
-    /// binary data.</summary>
+    /// <summary>Encodes an <see cref="IMessage" /> as a Protobuf length prefixed message into a
```

Comment:
I would simplify:
Encodes an IMessage as a length-prefixed message, using the Protobuf encoding.

"length-prefixed message" is actually a gRPC term, not a Protobuf term.